### PR TITLE
CI: Update ESLint Config

### PR DESCRIPTION
**Pull request by loicginoux**
## Summary

the es lint upgrade changed the way eslint config file is declared. we had to rename and migrate config options to the new configuration system

## Checklist

Check out our [PR guidelines](https://github.com/solidusio/.github/blob/master/CONTRIBUTING.md#pull-request-guidelines) for more details.

The following are mandatory for all PRs:

- [ ] I have written a thorough PR description.
- [ ] I have kept my commits small and atomic.
- [ ] [I have used clear, explanatory commit messages](https://github.com/solidusio/.github/blob/main/CONTRIBUTING.md#writing-good-commit-messages).

The following are not always needed:

- 📖 I have updated the README to account for my changes.
- 📑 I have documented new code [with YARD](https://www.rubydoc.info/gems/yard/file/docs/Tags.md).
- 🛣️ I have opened a PR to update the [guides](https://github.com/solidusio/edgeguides).
- ✅ I have added automated tests to cover my changes.
- 📸 I have attached screenshots to demo visual changes.

### CODE REVIEW

**Comment by tvdeyen** [comment]
@loicginoux nice! Thanks. I just disabled the new config in #5721 but this is probably better because its future proof :)

File: package.json [hunk 1]
```diff
@@ -0,0 +1,27 @@
+{
+  "name": "solidus",
+  "version": "0.0.1",
```
Comment (tvdeyen) [inline package.json:3]: I think we should not mimic to be a npm package.

Can we remove all of the config and reduce it to the bare minimum to satisfy eslint?

**Review by tvdeyen — DISMISSED** [review]
Nice, but we should not pretend to be a npm package. We just use the `package.json` for dependency management for ESLint (and maybe other JS tools), but never will publish Solidus as NPM package 😆

File: package.json (hunk 1 above)
Comment (loicginoux) [reply package.json:3]: I don't know another way to be able to use an external npm package (eslint, etc..) without a package.json file. A package.json file must contain at least a name and a version. What we could do is enforcing it to be private to make things clear with the `{"private": "true"}`. source [here](https://stackoverflow.com/questions/57365415/bare-minimum-package-json)

**Comment by tvdeyen** [comment]
It seems there is more involved. Maybe we merge https://github.com/solidusio/solidus/pull/5721 first and then continue the work here?

**Comment by loicginoux** [comment]
I'll continue exploring the CI environment, I guess it's just a missing `npm install` in CI env.

**Comment by loicginoux** [comment]
It looks good now @tvdeyen, I can squash commits if requested.

**Comment by tvdeyen** [comment]
I merged #5721 now. I am not sure we want this change as this means we need to maintain a `package.json` with its dependencies and a custom eslint config file. Before we where just using `npx` and the default eslint config.

**Comment by loicginoux** [comment]
I understand, fair point... I leave you close the issue if you think this problem is solved.

**Comment by tvdeyen** [comment]
@solidusio/core-team wdyt?

**Comment by kennyadsl** [comment]
I like the completeness of this solution, but considering that ESlint is a very very small part of this project, I'd avoid any maintenance effort for the plugin. I'd probably rather remove it and try to use something else. I think that for now the solution @tvdeyen proposed is enough, but I'd keep this as a reference if we want to change idea later. 

Thanks, @loicginoux and @tvdeyen !

**Comment by tvdeyen** [comment]
Thanks for working on this @loicginoux but i agree with @kennyadsl and am closing this for now. Again, thanks for investing and contributing.

**Pull request closed without merging**